auto3dseg/algorithm_templates/dints/scripts/infer.py

```diff
@@ -64,91 +66,88 @@ def get_mem_from_visible_gpus():
+                batch_size = int(batch_size)
+                batch_size = max(batch_size, 1)
+
+                parser["training"].update({"num_patches_per_iter": batch_size})
+                parser["training"].update({"num_crops_per_image": 2 * batch_size})
+
+                # estimate data size based on number of images and image
+                # size
+                _factor = 1.0
+
+                try:
+                    _factor *= 1251.0 / float(parser["stats_summary"]["n_cases"])
+                    _mean_shape = parser["stats_summary"]["image_stats"]["shape"]["mean"]
+                    _factor *= float(_mean_shape[0]) / 240.0
+                    _factor *= float(_mean_shape[1]) / 240.0
+                    _factor *= float(_mean_shape[2]) / 155.0
+                except BaseException:
+                    pass
+
+                _patch_size = parser["training"]["roi_size"]
+                _factor *= 96.0 / float(_patch_size[0])
+                _factor *= 96.0 / float(_patch_size[1])
+                _factor *= 96.0 / float(_patch_size[2])
+
+                if "training#epoch_divided_factor" in override:
+                    epoch_divided_factor = override["training#epoch_divided_factor"]
+                else:
+                    epoch_divided_factor = parser["training"]["epoch_divided_factor"]
+                epoch_divided_factor = float(epoch_divided_factor)
+                _factor /= epoch_divided_factor
 
-                    _estimated_epochs = 400.0
-                    _estimated_epochs *= _factor
+                _factor = max(1.0, _factor)
 
-                    parser["training"].update({"num_epochs": int(_estimated_epochs / float(batch_size))})
+                _estimated_epochs = 400.0
+                _estimated_epochs *= _factor
 
+                parser["training"].update({"num_epochs": int(_estimated_epochs / float(batch_size))})
+                rank = int(os.getenv("RANK", "0"))
+                if rank == 0:
                     ConfigParser.export_config_file(parser.get(), _file, fmt="yaml", default_flow_style=None)
 
-    return
+    return parser
 
 
 class InferClass:
     def __init__(self, config_file: Optional[Union[str, Sequence[str]]] = None, **override):
-        pre_operation(config_file, **override)
-        if dist.is_initialized():
-            dist.barrier()
-
         logging.basicConfig(stream=sys.stdout, level=logging.INFO)
 
         _args = _update_args(config_file=config_file, **override)
         config_file_ = _pop_args(_args, "config_file")[0]
-
+        config_file_ = [path for path in ensure_tuple(config_file_) if not (path.endswith("hyper_parameters.yaml") or Path(path).name.startswith('.'))]
         parser = ConfigParser()
         parser.read_config(config_file_)
+        parser_hyper = pre_operation(config_file, **override)
         parser.update(pairs=_args)
+        parser.update(pairs=parser_hyper.config)
```

Comment:
What is the difference between line #149 and line #150?

Reply:
The `parser_hyper` is used to update the override hyperparameters updated in `pre_operation`. We don't try to load the new hyperparameter.yaml since it may causes issue in multigpu env.

Reply:
I feel the original logic is to do pre_operation first updating config.yaml, and then add changes from override. So the order of line https://github.com/Project-MONAI/research-contributions/pull/149 and line https://github.com/Project-MONAI/research-contributions/issues/150 should be switched?

Reply:
updated.